web/src/components/subscribe-newsletter/subscribe-newsletter.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import StateTree from '../../stores/tree';
+import API from '../../services/api';
+import { connect } from 'react-redux';
+import { CheckIcon, OldPlayIcon, CautionIcon } from '../ui/icons';
+
+import './subscribe-newsletter.css';
+
+interface PropsFromState {
+  api: API;
+}
+
+interface State {
+  email: string;
+  buttonStyle: string;
+  isSubmitted: boolean;
+  isValidEmail: boolean;
+}
+
+class SignUp extends React.Component<PropsFromState, State> {
+  state = {
+    email: '',
+    buttonStyle: 'submit-button',
+    isSubmitted: false,
+    isValidEmail: true,
+  };
+
+  handleChange = (event: any) => {
+    this.setState({ email: event.target.value });
```

Comment:
You can set multiple state properties in one call, by putting them into the same object, e.g.: `setState({email: 'something', isSubmitted: false})`.

It's also a good pattern in React to not use derived state (unless it's costly to compute). In this case that would be isValidEmail (which can be derived from email) and buttonStyle (which can be derived from isValidEmail and isSubmitted).